src/shm_mod.c

```diff
@@ -558,8 +558,9 @@ sr_shmmod_modinfo_rdlock_upgrade(struct sr_mod_info_s *mod_info, uint32_t sid)
         mod = &mod_info->mods[i];
         shm_lock = &mod->shm_mod->data_lock_info[mod_info->ds];
 
-        /* upgrade only read-upgr-locked modules */
-        if (mod->state & MOD_INFO_RLOCK_UPGR) {
+        /* upgrade only required read-upgr-locked modules and leave others read-upgr-locked to prevent their locking causing potential dead-lock */
+        if ((mod->state & (MOD_INFO_RLOCK_UPGR | MOD_INFO_REQ)) ==
+            (MOD_INFO_RLOCK_UPGR | MOD_INFO_REQ)) {
```

Comment:
Here I think it should have one indent more.

Reply:
Ok, done. My distro doesn't have a version of uncrustify new enough for cmake to use.

Reply:
Neither does the Ubuntu 18.04 the CI is running on, it is compiled there from [source](https://github.com/sysrepo/sysrepo/blob/devel/.github/workflows/ci.yml#L127).